domain/servicefactory/model.go

```diff
@@ -167,6 +169,15 @@ func (s *ModelFactory) Secret(adminConfigGetter secretservice.BackendAdminConfig
 	)
 }
 
+// Agent returns the model's agent service.
+func (s *ModelFactory) Agent() *modelagentservice.ModelService {
```

Comment:
I'd rather be consistent and call this `ModelAgent`.

Reply:
@barrettj12 would you mind looking at this when you have 5?